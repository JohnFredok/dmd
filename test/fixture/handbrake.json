[
  {
    "id": "module:handbrake-js~Handbrake",
    "longname": "module:handbrake-js~Handbrake",
    "name": "Handbrake",
    "scope": "inner",
    "kind": "class",
    "augments": [
      "external:EventEmitter"
    ],
    "description": "A thin wrapper on the handbrakeCLI child_process handle. An instance of this class is returned by {@link module:handbrake-js.spawn}.",
    "memberof": "module:handbrake-js",
    "fires": [
      "module:handbrake-js~Handbrake#event:start",
      "module:handbrake-js~Handbrake#event:begin",
      "module:handbrake-js~Handbrake#event:progress",
      "module:handbrake-js~Handbrake#event:output",
      "module:handbrake-js~Handbrake#event:error",
      "module:handbrake-js~Handbrake#event:end",
      "module:handbrake-js~Handbrake#event:complete"
    ],
    "codeName": "Handbrake",
    "order": 0
  },
  {
    "id": "module:handbrake-js~Handbrake#output",
    "longname": "module:handbrake-js~Handbrake#output",
    "name": "output",
    "scope": "instance",
    "kind": "member",
    "description": "A `String` containing all handbrakeCLI output",
    "memberof": "module:handbrake-js~Handbrake",
    "type": {
      "names": [
        "string"
      ]
    },
    "codeName": "this.output",
    "order": 1
  },
  {
    "id": "module:handbrake-js~Handbrake#options",
    "longname": "module:handbrake-js~Handbrake#options",
    "name": "options",
    "scope": "instance",
    "kind": "member",
    "description": "the options HandbrakeCLI was spawned with",
    "memberof": "module:handbrake-js~Handbrake",
    "type": {
      "names": [
        "object"
      ]
    },
    "codeName": "this.options",
    "order": 2
  },
  {
    "id": "module:handbrake-js~Handbrake#event:start",
    "longname": "module:handbrake-js~Handbrake#event:start",
    "name": "start",
    "scope": "instance",
    "kind": "event",
    "description": "Fired as HandbrakeCLI is launched. Nothing has happened yet.",
    "memberof": "module:handbrake-js~Handbrake",
    "order": 3
  },
  {
    "id": "module:handbrake-js~Handbrake#event:begin",
    "longname": "module:handbrake-js~Handbrake#event:begin",
    "name": "begin",
    "scope": "instance",
    "kind": "event",
    "description": "Fired when encoding begins. If you're expecting an encode and this never fired, something went wrong.",
    "memberof": "module:handbrake-js~Handbrake",
    "order": 4
  },
  {
    "id": "module:handbrake-js~Handbrake#event:progress",
    "longname": "module:handbrake-js~Handbrake#event:progress",
    "name": "progress",
    "scope": "instance",
    "kind": "event",
    "description": "Fired at regular intervals passing a `progress` object.",
    "memberof": "module:handbrake-js~Handbrake",
    "params": [
      {
        "type": {
          "names": [
            "object"
          ]
        },
        "description": "details of encode progress",
        "name": "progress"
      },
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "current task index",
        "name": "progress.taskNumber"
      },
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "total tasks in the queue",
        "name": "progress.taskCount"
      },
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "name": "progress.percentComplete"
      },
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "Frames per second",
        "name": "progress.fps"
      },
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "Average frames per second",
        "name": "progress.avgFps"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "Estimated time until completion",
        "name": "progress.eta"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "Task description, either \"Encoding\" or \"Muxing\"",
        "name": "progress.task"
      }
    ],
    "order": 5
  },
  {
    "id": "module:handbrake-js~Handbrake#event:output",
    "longname": "module:handbrake-js~Handbrake#event:output",
    "name": "output",
    "scope": "instance",
    "kind": "event",
    "memberof": "module:handbrake-js~Handbrake",
    "params": [
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "An aggregate of `stdout` and `stderr` output from the underlying HandbrakeCLI process.",
        "name": "output"
      }
    ],
    "order": 6
  },
  {
    "id": "module:handbrake-js~Handbrake#event:error",
    "longname": "module:handbrake-js~Handbrake#event:error",
    "name": "error",
    "scope": "instance",
    "kind": "event",
    "memberof": "module:handbrake-js~Handbrake",
    "params": [
      {
        "type": {
          "names": [
            "Error"
          ]
        },
        "description": "All operational exceptions are delivered via this event.",
        "name": "error"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "One of `HandbrakeCLINotFound`, `HandbrakeCLIError`, `NoTitleFound`, `HandbrakeCLICrash` or `ValidationError`",
        "name": "error.name"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "name": "error.message"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "name": "error.errno"
      }
    ],
    "order": 7
  },
  {
    "id": "module:handbrake-js~Handbrake#event:end",
    "longname": "module:handbrake-js~Handbrake#event:end",
    "name": "end",
    "scope": "instance",
    "kind": "event",
    "description": "Fired on successful completion of an encoding task. Always follows a `begin` event, with some `progress` in between.",
    "memberof": "module:handbrake-js~Handbrake",
    "order": 8
  },
  {
    "id": "module:handbrake-js~Handbrake#event:complete",
    "longname": "module:handbrake-js~Handbrake#event:complete",
    "name": "complete",
    "scope": "instance",
    "kind": "event",
    "description": "Fired when HandbrakeCLI exited cleanly. This does not necessarily mean your encode completed as planned..",
    "memberof": "module:handbrake-js~Handbrake",
    "order": 9
  },
  {
    "id": "external:EventEmitter",
    "longname": "external:EventEmitter",
    "name": "EventEmitter",
    "scope": "global",
    "kind": "external",
    "see": [
      "http://nodejs.org/api/events.html"
    ],
    "order": 10
  },
  {
    "id": "module:handbrake-js",
    "longname": "module:handbrake-js",
    "name": "handbrake-js",
    "kind": "module",
    "description": "Handbrake for node.js.",
    "examples": [
      "```js\nvar hbjs = require(\"handbrake-js\");\n```"
    ],
    "typicalname": "hbjs",
    "order": 11
  },
  {
    "id": "module:handbrake-js.spawn",
    "longname": "module:handbrake-js.spawn",
    "name": "spawn",
    "scope": "static",
    "kind": "function",
    "description": "Spawns a HandbrakeCLI process with the supplied [options](https://trac.handbrake.fr/wiki/CLIGuide#options), returning an instance of `Handbrake` on which you can listen for events.",
    "memberof": "module:handbrake-js",
    "alias": "module:handbrake-js.spawn",
    "params": [
      {
        "type": {
          "names": [
            "Object"
          ]
        },
        "description": "[Options](https://trac.handbrake.fr/wiki/CLIGuide#options) to pass directly to HandbrakeCLI",
        "name": "options"
      }
    ],
    "examples": [
      "```js\nvar hbjs = require(\"handbrake-js\");\n\nhbjs.spawn(options)\n    .on(\"error\", console.error)\n    .on(\"output\", console.log);\n```"
    ],
    "returns": [
      {
        "type": {
          "names": [
            "module:handbrake-js~Handbrake"
          ]
        }
      }
    ],
    "codeName": "spawn",
    "order": 12
  },
  {
    "id": "module:handbrake-js.exec",
    "longname": "module:handbrake-js.exec",
    "name": "exec",
    "scope": "static",
    "kind": "function",
    "description": "Runs HandbrakeCLI with the supplied [options](https://trac.handbrake.fr/wiki/CLIGuide#options) calling the supplied callback on completion. The exec method is best suited for short duration tasks where you can wait until completion for the output.",
    "memberof": "module:handbrake-js",
    "alias": "module:handbrake-js.exec",
    "params": [
      {
        "type": {
          "names": [
            "Object"
          ]
        },
        "description": "[Options](https://trac.handbrake.fr/wiki/CLIGuide#options) to pass directly to HandbrakeCLI",
        "name": "options"
      },
      {
        "type": {
          "names": [
            "function"
          ]
        },
        "optional": true,
        "description": "If passed, `onComplete(err, stdout, stderr)` will be called on completion, `stdout` and `stderr` being strings containing the HandbrakeCLI output.",
        "name": "onComplete"
      }
    ],
    "examples": [
      "```js\nvar hbjs = require(\"handbrake-js\");\n\nhbjs.exec({ preset-list: true }, function(err, stdout, stderr){\n    if (err) throw err;\n    console.log(stdout);\n});\n```"
    ],
    "codeName": "exec",
    "order": 13
  }
]