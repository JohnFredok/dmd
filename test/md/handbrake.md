<a name="module_handbrake-js"></a>
##handbrake-js
Handbrake for node.js.

**Example**  
```js
var hbjs = require("handbrake-js");
```
* [handbrake-js](#module_handbrake-js)
  * [hbjs.cliOptions](#module_handbrake-js.cliOptions)
  * [hbjs.spawn(options)](#module_handbrake-js.spawn)
  * [hbjs.exec(options, [onComplete])](#module_handbrake-js.exec)
  * [class: hbjs~Handbrake](#module_handbrake-js..Handbrake)
    * [handbrake.output](#module_handbrake-js..Handbrake#output)
    * [handbrake.options](#module_handbrake-js..Handbrake#options)
    * [event: "start"](#module_handbrake-js..Handbrake#event_start)
    * [event: "begin"](#module_handbrake-js..Handbrake#event_begin)
    * [event: "progress"](#module_handbrake-js..Handbrake#event_progress)
    * [event: "output"](#module_handbrake-js..Handbrake#event_output)
    * [event: "error"](#module_handbrake-js..Handbrake#event_error)
    * [event: "end"](#module_handbrake-js..Handbrake#event_end)
    * [event: "complete"](#module_handbrake-js..Handbrake#event_complete)

<a name="module_handbrake-js.cliOptions"></a>
###hbjs.cliOptions
[Command-line-args](https://github.com/75lb/command-line-args) option definitions, useful when building a CLI.

**Type**: `array`  
<a name="module_handbrake-js.spawn"></a>
###hbjs.spawn(options)
Spawns a HandbrakeCLI process with the supplied [options](https://trac.handbrake.fr/wiki/CLIGuide#options), returning an instance of `Handbrake` on which you can listen for events.

**Params**

- options `Object` | `Array` - [Options](https://trac.handbrake.fr/wiki/CLIGuide#options) to pass directly to HandbrakeCLI  

**Returns**: <code>[Handbrake](#module_handbrake-js..Handbrake)</code> - A `Handbrake` instance  
**Example**  
```js
var hbjs = require("handbrake-js");

hbjs.spawn(options)
    .on("error", console.error)
    .on("output", console.log);
```
<a name="module_handbrake-js.exec"></a>
###hbjs.exec(options, [onComplete])
Runs HandbrakeCLI with the supplied [options](https://trac.handbrake.fr/wiki/CLIGuide#options) calling the supplied callback on completion. The exec method is best suited for short duration tasks where you can wait until completion for the output.

**Params**

- options `Object` - [Options](https://trac.handbrake.fr/wiki/CLIGuide#options) to pass directly to HandbrakeCLI  
- \[onComplete\] `function` - If passed, `onComplete(err, stdout, stderr)` will be called on completion, `stdout` and `stderr` being strings containing the HandbrakeCLI output.  

**Example**  
```js
var hbjs = require("handbrake-js");

hbjs.exec({ preset-list: true }, function(err, stdout, stderr){
    if (err) throw err;
    console.log(stdout);
});
```
<a name="module_handbrake-js..Handbrake"></a>
###class: hbjs~Handbrake
A thin wrapper on the handbrakeCLI child_process handle. An instance of this class is returned by `hbjs.spawn()`.

**Extends**: `EventEmitter`  
**Scope**: inner class of [handbrake-js](#module_handbrake-js)  
  * [class: hbjs~Handbrake](#module_handbrake-js..Handbrake)
    * [handbrake.output](#module_handbrake-js..Handbrake#output)
    * [handbrake.options](#module_handbrake-js..Handbrake#options)
    * [event: "start"](#module_handbrake-js..Handbrake#event_start)
    * [event: "begin"](#module_handbrake-js..Handbrake#event_begin)
    * [event: "progress"](#module_handbrake-js..Handbrake#event_progress)
    * [event: "output"](#module_handbrake-js..Handbrake#event_output)
    * [event: "error"](#module_handbrake-js..Handbrake#event_error)
    * [event: "end"](#module_handbrake-js..Handbrake#event_end)
    * [event: "complete"](#module_handbrake-js..Handbrake#event_complete)

<a name="module_handbrake-js..Handbrake#output"></a>
####handbrake.output
A `String` containing all handbrakeCLI output

<a name="module_handbrake-js..Handbrake#options"></a>
####handbrake.options
the options HandbrakeCLI was spawned with

<a name="module_handbrake-js..Handbrake#event_start"></a>
####event: "start"
Fired as HandbrakeCLI is launched. Nothing has happened yet.

<a name="module_handbrake-js..Handbrake#event_begin"></a>
####event: "begin"
Fired when encoding begins. If you're expecting an encode and this never fired, something went wrong.

<a name="module_handbrake-js..Handbrake#event_progress"></a>
####event: "progress"
Fired at regular intervals passing a `progress` object containing:

- taskNumber `Number` current task index
- taskCount `Number` total tasks in the queue
- percentComplete `Number`
- fps `Number` Frames per second
- avgFps `Number` Average frames per second
- eta `String` Estimated time until completion
- task `String` Task description, either "Encoding" or "Muxing"

<a name="module_handbrake-js..Handbrake#event_output"></a>
####event: "output"
An aggregate of `stdout` and `stderr` output from the underlying HandbrakeCLI process.

<a name="module_handbrake-js..Handbrake#event_error"></a>
####event: "error"
All operational exceptions are delivered via this event. Passes an `Error` instance, the `name` value of which will be one of:

- HandbrakeCLINotFound
- HandbrakeCLIError
- NoTitleFound
- HandbrakeCLICrash
- ValidationError

<a name="module_handbrake-js..Handbrake#event_end"></a>
####event: "end"
Fired on successful completion of an encoding task. Always follows a `begin` event, with some `progress` in between.

<a name="module_handbrake-js..Handbrake#event_complete"></a>
####event: "complete"
Fired when HandbrakeCLI exited cleanly. This does not necessarily mean your encode completed as planned..

