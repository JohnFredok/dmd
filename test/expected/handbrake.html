<!DOCTYPE html>
<html>
  <head>
    <title>Docs</title>
    <meta charset="utf8">
    <link rel="stylesheet" href="/html/docs.css">
    <link rel="stylesheet" href="/html/coy.css">
  </head>
  <body class="overflow-hidden">
    <div class="flexbox-col box-fill-viewport">
      <div class="row1 flex-fixed">
        <h1>API Reference</h1>
      </div>
      <div class="row2 flexbox-row">
        <div class="col1 horiz-space flex-fixed overflow-scroll">
          <h3>Modules</h3><ul><li>
  <a href="#module_handbrake-js">handbrake-js</a>
  
    <ul>
    <li>
  <a href="#module_handbrake-js.cliOptions">hbjs.cliOptions</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js.spawn">hbjs.spawn(options)</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js.exec">hbjs.exec(options, [onComplete])</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake">class: hbjs~Handbrake</a>
  
    <ul>
    <li>
  <a href="#new_module_handbrake-js..Handbrake◊">new Handbrake()</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#output">handbrake.output</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#options">handbrake.options</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#event_start">event: "start"</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#event_begin">event: "begin"</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#event_progress">event: "progress"</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#event_output">event: "output"</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#event_error">event: "error"</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#event_end">event: "end"</a>
  
</li>

  
    
    <li>
  <a href="#module_handbrake-js..Handbrake#event_complete">event: "complete"</a>
  
</li>
</ul>
  
</li>
</ul>
  
</li>
</ul>
          

          

          
        </div>
        <div class="col2 overflow-scroll flex-full">
        
          <h2><a name="module_handbrake-js">handbrake-js</a></h2>
<p>Handbrake for node.js.</p>





<strong>Example</strong>
<pre><code class="language-javascript">var hbjs = require(&quot;handbrake-js&quot;);
</code></pre>



<dl>
  <dt>kind</dt><dd>module</dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <dt>typicalname</dt><dd>hbjs</dd>
  
  
</dl>

<h3><a name="module_handbrake-js.cliOptions">hbjs.cliOptions</a></h3>
<p><a href="https://github.com/75lb/command-line-args">Command-line-args</a> option definitions, useful when building a CLI.</p>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>static</dd>
  <dt>memberof</dt><dd><code>static member of <a href="#module_handbrake-js">handbrake-js</a></code></dd>
  
  <dt>type</dt><dd>array</dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h3><a name="module_handbrake-js.spawn">hbjs.spawn(options) -> [Handbrake](#module_handbrake-js..Handbrake)</a></h3>
<p>Spawns a HandbrakeCLI process with the supplied <a href="https://trac.handbrake.fr/wiki/CLIGuide#options">options</a>, returning an instance of <code>Handbrake</code> on which you can listen for events.</p>


<strong>Params</strong>
<ul>
<li>options <code>Object,Array</code> <a href="https://trac.handbrake.fr/wiki/CLIGuide#options">Options</a> to pass directly to HandbrakeCLI</li>
</ul>



<strong>Example</strong>
<pre><code class="language-javascript">var hbjs = require(&quot;handbrake-js&quot;);

hbjs.spawn(options)
    .on(&quot;error&quot;, console.error)
    .on(&quot;output&quot;, console.log);
</code></pre>



<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>static</dd>
  <dt>memberof</dt><dd><code>static function of <a href="#module_handbrake-js">handbrake-js</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h3><a name="module_handbrake-js.exec">hbjs.exec(options, [onComplete])</a></h3>
<p>Runs HandbrakeCLI with the supplied <a href="https://trac.handbrake.fr/wiki/CLIGuide#options">options</a> calling the supplied callback on completion. The exec method is best suited for short duration tasks where you can wait until completion for the output.</p>


<strong>Params</strong>
<ul>
<li>options <code>Object</code> <a href="https://trac.handbrake.fr/wiki/CLIGuide#options">Options</a> to pass directly to HandbrakeCLI</li>

<li>onComplete <code>function</code> If passed, <code>onComplete(err, stdout, stderr)</code> will be called on completion, <code>stdout</code> and <code>stderr</code> being strings containing the HandbrakeCLI output.</li>
</ul>



<strong>Example</strong>
<pre><code class="language-javascript">var hbjs = require(&quot;handbrake-js&quot;);

hbjs.exec({ preset-list: true }, function(err, stdout, stderr){
    if (err) throw err;
    console.log(stdout);
});
</code></pre>



<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>static</dd>
  <dt>memberof</dt><dd><code>static function of <a href="#module_handbrake-js">handbrake-js</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h3><a name="module_handbrake-js..Handbrake">class: hbjs~Handbrake</a></h3>
<p>A thin wrapper on the handbrakeCLI child_process handle. An instance of this class is returned by <code>hbjs.spawn()</code>.</p>






<dl>
  <dt>kind</dt><dd>class</dd>
  <dt>scope</dt><dd>inner</dd>
  <dt>memberof</dt><dd><code>inner class of <a href="#module_handbrake-js">handbrake-js</a></code></dd>
  <dt>augments</dt><dd>EventEmitter</dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="new_module_handbrake-js..Handbrake◊">new Handbrake()</a></h4>






<dl>
  <dt>kind</dt><dd>constructor</dd>
  
  <dt>memberof</dt><dd><code> constructor of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#output">handbrake.output</a></h4>
<p>A <code>String</code> containing all handbrakeCLI output</p>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance member of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#options">handbrake.options</a></h4>
<p>the options HandbrakeCLI was spawned with</p>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance member of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#event_start">event: "start"</a></h4>
<p>Fired as HandbrakeCLI is launched. Nothing has happened yet.</p>






<dl>
  <dt>kind</dt><dd>event</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance event of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#event_begin">event: "begin"</a></h4>
<p>Fired when encoding begins. If you&#39;re expecting an encode and this never fired, something went wrong.</p>






<dl>
  <dt>kind</dt><dd>event</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance event of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#event_progress">event: "progress"</a></h4>
<p>Fired at regular intervals passing a <code>progress</code> object containing:</p>
<ul>
<li>taskNumber <code>Number</code> current task index</li>
<li>taskCount <code>Number</code> total tasks in the queue</li>
<li>percentComplete <code>Number</code></li>
<li>fps <code>Number</code> Frames per second</li>
<li>avgFps <code>Number</code> Average frames per second</li>
<li>eta <code>String</code> Estimated time until completion</li>
<li>task <code>String</code> Task description, either &quot;Encoding&quot; or &quot;Muxing&quot;</li>
</ul>






<dl>
  <dt>kind</dt><dd>event</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance event of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#event_output">event: "output"</a></h4>
<p>An aggregate of <code>stdout</code> and <code>stderr</code> output from the underlying HandbrakeCLI process.</p>






<dl>
  <dt>kind</dt><dd>event</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance event of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#event_error">event: "error"</a></h4>
<p>All operational exceptions are delivered via this event. Passes an <code>Error</code> instance, the <code>name</code> value of which will be one of:</p>
<ul>
<li>HandbrakeCLINotFound</li>
<li>HandbrakeCLIError</li>
<li>NoTitleFound</li>
<li>HandbrakeCLICrash</li>
<li>ValidationError</li>
</ul>






<dl>
  <dt>kind</dt><dd>event</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance event of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#event_end">event: "end"</a></h4>
<p>Fired on successful completion of an encoding task. Always follows a <code>begin</code> event, with some <code>progress</code> in between.</p>






<dl>
  <dt>kind</dt><dd>event</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance event of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_handbrake-js..Handbrake#event_complete">event: "complete"</a></h4>
<p>Fired when HandbrakeCLI exited cleanly. This does not necessarily mean your encode completed as planned..</p>






<dl>
  <dt>kind</dt><dd>event</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance event of <a href="#module_handbrake-js..Handbrake">Handbrake</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>


        </div>
      </div>
    </div>
    <script src="/html/prism.js"></script>
  </body>
</html>
